app/models/report.rb

```diff
@@ -7,11 +7,29 @@ class Report < ApplicationRecord
   validates :title, presence: true
   validates :content, presence: true
 
+  has_many :mentions, dependent: :destroy
+  has_many :mentioning_reports, through: :mentions, source: :mentioned_report
+  has_many :reverse_of_mentions, class_name: 'Mention', foreign_key: 'mentioned_report_id', dependent: :destroy,
+                                 inverse_of: 'mentioned_report'
+  has_many :mentioned_reports, through: :reverse_of_mentions, source: :report
+  after_save :create_mention
+
   def editable?(target_user)
     user == target_user
   end
 
   def created_on
     created_at.to_date
   end
+
+  def create_mention
+    ids = content.scan(%r{http://localhost:3000/reports/(\d+)}).flatten.uniq
+    mentioned_report_ids = Report.where(id: ids).where.not(id:).pluck(:id)
+    mentions.destroy_all
```

Comment:
```suggestion
    mentions.destroy_all
    ids = content.scan(%r{http://localhost:3000/reports/(\d+)}).flatten.uniq
    mentioned_report_ids = Report.where(id: ids).where.not(id:).pluck(:id)
```

変数の寿命を短くする観点からいうと、destroyは最初に持ってきた方がいいかも
https://bootcamp.fjord.jp/pages/311